src/com/google/javascript/jscomp/TypeInference.java

```diff
@@ -134,6 +136,7 @@
     this.bottomScope =
         LinkedFlowScope.createEntryLattice(
             compiler, TypedScope.createLatticeBottom(syntacticScope.getRootNode()));
+    this.moduleTranspilationBeforeTypeInference = compiler.getOptions().processCommonJSModules;
```

Comment:
We try to avoid having classes other than Compiler base their behavior directly on Compiler options.
Instead, we prefer to have a boolean constructor parameter.
In this case, you'll need to also add one to `TypeInferencePass`, since that's where this constructor gets called.

Reply:
I wondered about that as I wrote it.

I've made the changes, but it affected more code than it probably should have. The only purpose of this is to prevent having to traverse the AST if a MODULE_BODY isn't found. I'm wondering now if this is even needed. What do you think?

Reply:
After reviewing it again, this just seemed like a premature optimization. I removed the explicit test and left this as the fallback case.